lambeq/rewrite/base.py

```diff
@@ -393,3 +419,34 @@ def _ar(self, box: Box) -> Diagram:
 
     def _ob(self, ob: Ty) -> Ty:
         return ob
+
+def HandleUnknownWords(diagrams: Container[Diagram], training_data: Container[str] | None = None, test_data: Container[str] | None = None, train_status: bool | None = True, min_freq: int | None = 0) -> Container[Diagram]:
```

Comment:
It would make more sense for `HandleUnknownWords` to be a class that has separate methods for training and test data, and can possibly store the vocabulary calculated during train time, this current method is fairly convoluted.